File: 0714 - Best Time to Buy and Sell Stock with Transaction Fee/solution.c
#define max(a,b) (((a)>(b))?(a):(b))
int maxProfit(int* prices, int pricesSize, int fee){
    int s1 = 0, s2 = -prices[0], ns1, ns2;
    for(int i = 1; i < pricesSize; i++){
        ns1 = max(s1, s2+prices[i]-fee);
        ns2 = max(s2, s1-prices[i]);
        s1 = ns1, s2 = ns2;
    }
    return s1;
}

File: 0714 - Best Time to Buy and Sell Stock with Transaction Fee/solution.py
class Solution:
    def maxProfit(self, prices: List[int], fee: int) -> int:
        s1, s2 = 0, -prices[0]
        for i in prices:
            s1, s2 = max(s1, s2+i-fee), max(s2, s1-i)
        return s1
        